examples/llm/yamls/mosaic_gpt/finetune/1b_local_data_sft.yaml

```diff
@@ -0,0 +1,128 @@
+tokenizer_name: gpt2 # This must match the tokenizer you pretrained with
+max_seq_len: 2048
+global_seed: 17
+
+# Run Name
+run_name: # If left blank, will be read from env var $COMPOSER_RUN_NAME
+
+# Model
+# These must match pretraining
+model:
+  name: mosaic_gpt
+  init_device: meta
+  tokenizer_name: ${tokenizer_name}
+  d_model: 2048
+  n_heads: 16 # Modified 24->16 so that d_head == 128 to statisfy FlashAttention
+  n_layers: 24
+  mlp_ratio: 4
+  max_seq_len: ${max_seq_len}
+  vocab_size: 50368
+  attn_impl: triton
+
+# Tokenizer
+tokenizer:
+  name: ${tokenizer_name}
+  kwargs:
+    model_max_length: ${max_seq_len}
+
+# Local data to load into huggingface datasets
+dataset: &hf_dataset
+  name: mylocalset
+  kwargs:
+    data_files:
+      train: /path/to/train.jsonl
+      test: /path/to/test.jsonl
+
+# Dataloaders
+train_loader: &train_loader
+  name: finetuning
+  dataset:
+    <<: *hf_dataset
+    split: train
+    tokenizer_name: ${tokenizer_name}
+    max_seq_length: ${max_seq_len}
+    allow_pad_trimming: false
+    decoder_only_format: true
+    packing_ratio: 3.5
+    shuffle: true
+  drop_last: true
+  num_workers: 8
+  pin_memory: false
+  prefetch_factor: 2
+  persistent_workers: true
+  timeout: 0
+
+eval_loader:
+  <<: *train_loader
```

Comment:
what is this sorcery
❤️

Reply:
@samhavens has discovered some YAML dark magics. Life will never be the same.

Reply:
Yes! We already are using YAML anchors and aliases for strings, but it turns out you can use them for dicts and lists as well!
https://joshdevlin.com/blog/yaml-repeating-sections/

Reply:
actually here is a more authoritative source https://yaml.org/type/merge.html